HW_10/api.py

```diff
@@ -0,0 +1,268 @@
+                self.update_id = None
+
+    def run_bot(self):
+        """Запуск бота"""
+        while True:
+            response = json.loads(
+                requests.get(f'{self.url}bot{self.token}/getupdates?timeout=10&offset={self.update_id}').text)
+            if response['ok']:
+                if response['result']:
+                    for msg in response['result']:
+                        chat_id = msg['message']['chat']['id']
+                        text = msg['message']['text']
+                        if self.chats.get(chat_id):
+                            self.chats[chat_id].processing_message(text)
+                        else:
+                            first_name = msg['message']['chat']['first_name']
+                            chat = Chat(chat_id, first_name, self.url, self.token)
+                            self.chats[chat_id] = chat
+                            chat.processing_message(text)
+                    self.update_id = response['result'][-1]['update_id'] + 1
+
+    @staticmethod
+    def send_bottom(chat_id, msg, text_bottom: tuple, url, token):
+        """Отправляет в чат кортеж кнопок"""
+        headers = {'Content-Type': 'application/json'}
+        keyword = []
+        key_line = []
+        for text in text_bottom:
+            key_line.append({"text": text})
+            if len(key_line) == 2:
+                keyword.append(key_line)
+                key_line = []
+        if key_line:
+            keyword.append(key_line)
+
+        data = {"chat_id": str(chat_id),
+                "text": msg,
+                "parse_mode": "html",
+                "reply_markup": {
+                    "keyboard": keyword,
+                    "resize_keyboard": True,
+                    "one_time_keyboard": True}}
+        requests.post(f'{url}bot{token}/sendmessage?', data=json.dumps(data), headers=headers)
+
+    @staticmethod
+    def send_message_html(chat_id, text, url, token):
+        """Отправляет в чат сообщение"""
+        headers = {'Content-Type': 'application/json'}
+        data = {"chat_id": str(chat_id), "text": text, "parse_mode": "html"}
+        requests.post(f'{url}bot{token}/sendmessage?', data=json.dumps(data), headers=headers)
+
+    @staticmethod
+    def send_message__(chat_id, text, url, token):
+        """Отправляет в чат сообщение"""
+        headers = {'Content-Type': 'application/json'}
+        data = {"chat_id": str(chat_id), "text": text}
+        requests.post(f'{url}bot{token}/sendmessage?', data=json.dumps(data), headers=headers)
+
+
+class Chat:
+    """Класс чата"""
+    phone_book = False
+    addition_polynomials = False
+    add_rec = False
+    add_first_name = False
+    add_last_name = False
+    add_tel = False
+    add_birthday = False
+    add_workplace = False
+    edit_rec = False
+    get_id = False
+    id_rec = None
+    search_last_name = False
+    res_polynom = None
+    operation = ("Телефонный_справочник", "Сложение_многочленов")
```

Comment:
Чтобы создать кортеж, необязательно использовать скобки, достаточно просто запятых. Соответственно, чтобы создать кортеж с одним элементом, просто поставьте после него одну запятую.
```suggestion
    operation = "Телефонный_справочник", "Сложение_многочленов"
```